terraform/lambda/main.tf

```diff
@@ -0,0 +1,241 @@
+        ]
+        Resource = [
+          "arn:aws:ecr:${var.region}:${var.aws_account_id}:repository/${var.ecr_repository_name}",
+          "arn:aws:ecr:${var.region}:${var.aws_account_id}:repository/${var.ecr_repository_name}/*"
+        ]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "ecr:GetAuthorizationToken"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# 4. Add S3 access policy
+resource "aws_iam_role_policy" "lambda_s3_access" {
+  name = "${var.domain}-${var.service_subdomain}-lambda-s3-policy"
+  role = aws_iam_role.lambda_execution_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:GetObject",
+          "s3:ListBucket"
+        ]
+        Resource = [
+          "arn:aws:s3:::${var.source_bucket}",
+          "arn:aws:s3:::${var.source_bucket}/*"
+        ]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:GetObject",
+          "s3:PutObject",
+          "s3:ListBucket"
+        ]
+        Resource = [
+          "arn:aws:s3:::${var.destination_bucket}",
+          "arn:aws:s3:::${var.destination_bucket}/*"
+        ]
+      }
+    ]
+  })
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# 5. Add additional permissions
+resource "aws_iam_role_policy" "lambda_additional_permissions" {
+  name = "${var.domain}-${var.service_subdomain}-policy-2"
+  role = aws_iam_role.lambda_execution_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "secretsmanager:GetSecretValue"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup"
+        ]
+        Resource = "arn:aws:logs:${var.region}:${var.aws_account_id}:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = [
+          "arn:aws:logs:${var.region}:${var.aws_account_id}:log-group:/aws/lambda/${var.domain}-${var.service_subdomain}-lambda:*"
+        ]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "execute-api:Invoke",
+          "execute-api:ManageConnections"
+        ]
+        Resource = "arn:aws:execute-api:*:*:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:*",
+          "s3-object-lambda:*"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# 6. Create the Lambda function
+resource "aws_lambda_function" "tech_audit_lambda" {
```

Comment:
This needs to sit in the private vpc. Example here: https://github.com/ONS-Innovation/github-copilot-usage-dashboard/blob/5121be17233c8ae9c0c43655c5c3a59ddcefce8b/terraform/data_logger/main.tf#L34

Reply:
I think what I've done works.

Reply:
Can confirm, its in the private VPC. You can view this yourself by going to the lambda > Configuration > VPC.